### ISSUE TITLE
Make variable, argument naming etc. configurable

### ISSUE THREAD

**Issue by timbeadle** [cflint/CFLint#492]
It would be good to be able to configure the various rules around naming/case, e.g.

* `VAR_INVALID_NAME`,
* `ARGUMENT_INVALID_NAME`,
* `METHOD_INVALID_NAME `,
* `COMPONENT_INVALID_NAME`

I would like to specify that (e.g.) variables, arguments and components are `PascalCase`, while methods are `camelCase`. This is fairly standard in linters in other languages, e.g. Javascript or SCSS.

(As an aside, the message says "Should be CamelCase or use underscores", but this should say "Should be camelCase or use underscores" as that's what currently being asked for.)

Also: configurability on some of the other rules would be good. I would like to specify what constitutes a too wordy, too long or too short variable/argument/method/component name.

e.g.

* `VAR_MIN_LENGTH: 2`
* `VAR_MAX_LENGTH: 20`

Rulesets could be published as sets of presets, similar to eslint's presets - AirBnb, Google etc.

**Comment by ryaneberly** [comment]
Fixed.  you can configure camel v pascal with the following .cflintrc:
```
{
   "parameters" : {
       "ComponentNameChecker.case" : "PascalCase"
   }
}
```

The same applies to most other parameters.

**Comment by TheRealAgentK** [comment]
Is there a list of all the parameters supported somewhere for documentation purposes?

**Comment by ryaneberly** [comment]
@TheRealAgentK 
How does this look:
https://github.com/cflint/CFLint/blob/documen/RULES.md

**Comment by bardware** [comment]
> How does this look

404 :(

**Comment by KamasamaK** [comment]
@bardware It was already merged, so that branch is gone. [Here](https://github.com/cflint/CFLint/blob/master/RULES.md).

**Comment by TheRealAgentK** [comment]
Cool, @ryaneberly - did you create it manually or via some automated output based on the source?

**Comment by ryaneberly** [comment]
@TheRealAgentK .  It's automated.   java -jar CFLint-xxx.jar -markdown
